app/lib/common/utilities/genome_data.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:http/http.dart';
+
+import '../../profile/models/hive/alleles.dart';
+import '../../profile/models/hive/cpic_lookup_response.dart';
+import '../../profile/models/hive/diplotype.dart';
+import '../constants.dart';
+import '../services.dart';
+
+Future<void> fetchAndSaveAllesData(String token, String url) async {
+  final userAlleleData = getBox<Alleles>(Boxes.alleles);
+  if (userAlleleData.get('alleles') == null) {
+    final response = await getStarAlleles(token, url);
+    if (response.statusCode == 200) {
+      await _saveAlleleData(response);
+    } else {
+      throw Exception('Error occurred during loading of allele data');
+    }
+  }
+}
+
+Future<Response> getStarAlleles(String? token, String url) async =>
+    get(Uri.parse(url), headers: {'Authorization': 'Bearer $token'});
```

Comment:
```suggestion
Future<Response> getStarAlleles(String? token, String url) async {
        return get(Uri.parse(url), headers: {'Authorization': 'Bearer $token'});
}
```

I prefer this notation if the function body cant fit on the same line as the signature (the identation style messes with prettier sometimes)